### DISCUSSION TITLE
SearX/SearXng; Instances list; Engines list

### DISCUSSION THREAD

**Discussion by bphd** [searxng/searxng#1698] [Q&A] [answered]
Is SearXng is different from SearX? If yes, why don't you **detail** precisely the **differences on the ReadMe**, and why you redirect to SearX.Space to use SearXng without further instructions, considering that it's a place where all types of SearX instances are listed, vanilla or not? Wouldn't it be better to redirect to a **place where SearXng instances exclusively** are hosted? Or just telling how to **filter SearXng** in the SearX.Space?

Because personally I search for a kind of "ultimate" search engine, who would give me results from many engines, a little bit like AllTheInternet, AllIO, SearchItAll... But for real, not just a place to open tons of tabs. **A** real **search bar** with **one page of results** of these search engines. I get that it's difficult to retreive and sort them, but even and old cached version sorted randomly would be better for me than having to go to all search engines who exist

Another question. Why it is listed nowhere the famous **70+ engines that it uses?** By searching I only find a py script and a bang list, both of them telling me nothing if it search effectively in all those places, and what places does it really retreive

If I'm wrong and SearXng doesn't do what I want. Then I would ask to **be more precise in the ReadMe** about what is SearXng. And maybe if you know, to **redirect me** to something who do it

**Comment by return42** [comment, marked as answer]
> Is SearXng is different from SearX?

constant development, a more active community

> If yes, why don't you detail precisely 

SearXNG stands for its own .. its not our job to track searx development or the development of the many other forks and update a list of differences.

TBH: we do track searx and other forks, just to see if there are activities we can take over .. in practices SearXNG is much more active and we very rare pick from other forks.

> Or just telling how to filter SearXng in the SearX.Space?

We host and maintain searx.space and all the other services around (searx is only searx-sources) .. we consider to drop searx instance from the list.

> Because personally I search for a kind of "ultimate" search engine, who would give me results from many engines,

this is the intention of a metasearch engine and SearXNG is the front-runner in this.

> Why it is listed nowhere the famous 70+ engines that it uses? 

Visit our docs https://docs.searxng.org/admin/engines/configured_engines.html or at any instance click on "preferences" and select tab "search engine" https://darmarit.org/searx/preferences

> If I'm wrong and SearXng doesn't do what I want. 

Give it a try and you will love it. :+1:

**Reply by bphd** [reply]
Okay, thanks to your answers I understand better

I thought that SearXng was an alternative using more engines. But it's just a continuation of what seems to be an almost abandonware that is SearX. Or maybe it was different visions of how it should evolve

About the engines I would have preferred to have them listed on the ReadMe, because even on the official website I don't find them. But they're on your link

About the bangs, it's not clear, like, at all. I don't understand at all if the parameters enable the usage of the bangs or not. Because wether I activate the engine or not, the bang doesn't redirect to the website. It just seem to prioritize results. Which is a good idea for privacy. But is a problem for websites giving services, not just results. Like Google translate (!gten, !gtes, !gtde...) who gives you immediate translation, or WolfRam Alpha giving you complex results. DuckDuckHack can help about that to stay on the same engine, maybe not SearXng. So maybe some bang should redirect, and some other should filter the results

And moreover, it wasn't clear at all that those option were activating search engines and not bangs. It's probably my fault because I'm used to get a rapid setup tutorial on the ReadMe of the repo

Question, haven't you already tried, you or a fork, to ask to supported engines to be more permissible about SearX requests? Because a lot of them are blocked as spam with Captcha ou just time ban. For example, partiuclarly Mojeek and StartPage gives a lot of denial. It's funny for StartPage, who is himself a metaengine requesting other engines and is open source. It's like returning against it's own existance

I think it should be a really good idea to distingguish between primary sources of information and secondary. To distinguish in the settings between meta search engines who will make a dupe search (like Google and StartPage ultimately asking both to Google database) and search engines who claims to take results from their own crawl of the web, like Alexa or Google... That will give a better image of SearX, because it can be considered a spam engine by making multiple time the same request on the same database by different means

And by default, at instance installation. Only primary source should be activated, and optionally seocndary (meta engines) could be activated

And I think there should be preconfiguration. Like a button under the search engine "search with configuration 1" (for exaple I would have set Google only on configuration 1) and a button "search with configuraiton 2" Here I would have set a lot of search engines to get a broad and slow answer. And configure the default configuration that would have been the one to use with the browser omnibar

**Comment by return42** [comment]
> About the bangs, it's not clear, like, at all.

We have two where other only have one :-) .. "bang" and "external bang" .. visit info page on the instance https://darmarit.org/searx/info/en/search-syntax or have a look at the user-docs https://docs.searxng.org/user/index.html

> And by default, at instance installation. Only primary source should be activated, and optionally seocndary (meta engines) could be activated

SearXNG folds equal hits together.  I think we have a good mix of response time of search engines and search engines in the categories.

> It's probably my fault because I'm used to get a rapid setup tutorial on the ReadMe of the repo

The README is just to give an overview its not intended for users, there are much more aspects when you want to host your own instance .. we cover in our (verbose) documentation: https://docs.searxng.org/

----

Don't worry so much, just use SearXNG first to gain experience in using it.  Make your search queries in the general-, the images- or e.g. in the video-category.  That's how 90% of users use it / we try to provide best UX without having to know much about SearXNG (its made for "dumb users").  If you want more, read something about the search-syntax in the "about" page and try the suggestions made there.

If you have any questions or want to discuss, there is a large community on Matrix: https://matrix.to/#/#searxng:matrix.org.

SearXNG is primarily designed to protect your privacy, when choosing a search engine it is always a question of which instance provider you trust.

You can make first experiences with my instance https://darmarit.org/searx/ or you can use one of the instances of contributors, here I can recommend the following:

- https://searx.be/
- https://searx.tiekoetter.com/
- https://paulgo.io/

But you can also choose one at https://searx.space/ ... pay less attention to the ranking but more to the actuality (in the column "version").  With current versions it can be assumed that the operator also maintains his instance regularly.  If version starts with "1.x.x" ist not an SearXNG instance (we consider to drop them from the list, as I said before).

Don't use instance with a red background in the "Network" column .. we mark instances "unsecure" which are hosted on CDNs known to be a "man in the middle" / e.g. Cloudflare or AMAZON-02

**Reply by bphd** [reply]
> > About the bangs, it's not clear, like, at all.
> 
> We have two where other only have one :-) .. "bang" and "external bang" .. visit info page on the instance https://darmarit.org/searx/info/en/search-syntax or have a look at the user-docs https://docs.searxng.org/user/index.html
> 
> > And by default, at instance installation. Only primary source should be activated, and optionally seocndary (meta engines) could be activated
> 
> SearXNG folds equal hits together. I think we have a good mix of response time of search engines and search engines in the categories.
> 
> > It's probably my fault because I'm used to get a rapid setup tutorial on the ReadMe of the repo
> 
> The README is just to give an overview its not intended for users, there are much more aspects when you want to host your own instance .. we cover in our (verbose) documentation: https://docs.searxng.org/
> 
> Don't worry so much, just use SearXNG first to gain experience in using it. Make your search queries in the general-, the images- or e.g. in the video-category. That's how 90% of users use it / we try to provide best UX without having to know much about SearXNG (its made for "dumb users"). If you want more, read something about the search-syntax in the "about" page and try the suggestions made there.
> 
> If you have any questions or want to discuss, there is a large community on Matrix: https://matrix.to/#/#searxng:matrix.org.
> 
> SearXNG is primarily designed to protect your privacy, when choosing a search engine it is always a question of which instance provider you trust.
> 
> You can make first experiences with my instance https://darmarit.org/searx/ or you can use one of the instances of contributors, here I can recommend the following:
> 
>     * https://searx.be/
> 
>     * https://searx.tiekoetter.com/
> 
>     * https://paulgo.io/
> 
> 
> But you can also choose one at https://searx.space/ ... pay less attention to the ranking but more to the actuality (in the column "version"). With current versions it can be assumed that the operator also maintains his instance regularly. If version starts with "1.x.x" ist not an SearXNG instance (we consider to drop them from the list, as I said before).
> 
> Don't use instance with a red background in the "Network" column .. we mark instances "unsecure" which are hosted on CDNs known to be a "man in the middle" / e.g. Cloudflare or AMAZON-02

It folds equal hits, but it stills make those hits. Who slow down searches and make it appear as spam engine to other engines

My critieria to select were the country (to get less ping, but yeah sometimes it doesn't mean anything with good CDN) and reply time to a default seach. I haven't found any other relevant criteria because they all seem to have the same configurability, interface, all. I'm lucky, the one I choose was really up to date

Whats the use of a maintainer that maintain his instance regularly (besides having bleeding edge functions and maybe a faster service (but thats not all times correlated)? Because I haven't searched after that

**Reply by return42** [reply]
> It folds equal hits, but it stills make those hits. Who slow down searches and make it appear as spam engine to other engines

If you are not satisfied with the presets, you can activate/deactivate the engines according to your wishes (in the preferences).  Or you can run your own instance and select the presets yourself.

In any case, you should first gather extensive experience to evaluate which settings might be more advantageous.

**Reply by bphd** [reply]
> > It folds equal hits, but it stills make those hits. Who slow down searches and make it appear as spam engine to other engines
> 
> If you are not satisfied with the presets, you can activate/deactivate the engines according to your wishes (in the preferences). Or you can run your own instance and select the presets yourself.
> 
> In any case, you should first gather extensive experience to evaluate which settings might be more advantageous.

Thats why I advise in the UI to create two column, one for primary (crawler) engines and a second for secondary (metaengines) ones. But sure, if I learn the backend of every search engines, I could manually make a dynamic chart to extrapolate every dupe query and create a list who only ask different crawler databases. But you know, limited lifetime

I don't understand the hostname replace plugin. I guess that it replace things like "reddit.com" to "old.reddit.com", but it's really not explained well in the legend of the configuration page. Maybe it should be improved. And more, I don't see where I should configure such replacement and what replacement per default he does

Another suggestion, DuckDuck hacks are really avanced now, maybe better than google snippets. I think SearXng should win a lot working to make them compatible with the engine. Because for now there isn't a lot of instant answer plugin native
